### DISCUSSION TITLE
Is it possible to automate SSL import

### DISCUSSION THREAD

**Discussion by Tmac013GitHub** [nextgenhealthcare/connect#4737] [Q&A]
Is it possible to automate SSL import?  I found a powershell script that successfully connects to MIRTH and there is some SSL code but nothing to implement automation of selecting and importing SSL certificate.

**Comment by jonbartels** [comment]
Are you already using Connect with an SSL extension from Zen or from NextGen? If yes each of those does expose an API (possibly undocumented, since its meant for internal use by the extension client) to manage SSL.

If no, to where do you want to import certificates to for MC? Your only option when not using an SSL extension would be to load those to the cacerts truststore for the JVM that runs MC. No API exists for that, you just have to manage the truststore.

**Reply by pacmano1** [reply]
The swagger docs show:

![image](https://user-images.githubusercontent.com/44065187/134964877-dd25e3b6-faaa-4be4-a82e-fce66a500037.png)

Thanks to @tonygermano for telling me to look closer at the swagger docs.

**Reply by Tmac013GitHub** [reply]
Do you know how to implement the POST  /extensions/ssl/all API?  The powershell script that I found does not show how and the author's comment tells that he was not able to make it work.

**Reply by pacmano1** [reply]
I don't.  The swagger docs are quite often wrong.   I'll look at it later.

**Comment by Tmac013GitHub** [comment]
Using Mirth Adminstration utility to import SSL; However, the user wants to automate this process and bypass launching the Admin, selecting "Settings", click on "SSL", etc.  

Sent from Yahoo Mail on Android 
 
  On Tue, Sep 21, 2021 at 5:54 PM, ***@***.***> wrote:   


Are you already using Connect with an SSL extension from Zen or from NextGen? If yes each of those does expose an API (possibly undocumented, since its meant for internal use by the extension client) to manage SSL.

If no, to where do you want to import certificates to for MC? Your only option when not using an SSL extension would be to load those to the cacerts truststore for the JVM that runs MC. No API exists for that, you just have to manage the truststore.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by Tmac013GitHub** [comment]
Thanks Pacman!  I'll try it out.

Sent from Yahoo Mail on Android 
 
  On Mon, Sep 27, 2021 at 2:26 PM, ***@***.***> wrote:   


The swagger docs show:



—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
